src/daemon/osd_scenarios/osd_volume_activate.sh

```diff
@@ -7,13 +7,16 @@ function osd_volume_activate {
   # Find the OSD FSID from the OSD ID
   OSD_FSID="$(ceph-volume lvm list --format json | python -c "import sys, json; print(json.load(sys.stdin)[\"$OSD_ID\"][0][\"tags\"][\"ceph.osd_fsid\"])")"
 
+  # Find the OSD type
+  OSD_TYPE="$(ceph-volume lvm list --format json | python -c "import sys, json; print(json.load(sys.stdin)[\"$OSD_ID\"][0][\"type\"])")"
```

Comment:
That makes a 2nd call to c-v lvm list with almost the same code.
I would suggest the following.
make a single c-v lvm list call and store the result. Then parse it to extract your two fields while checking they actually exist before printing to avoid a assert.
Why not having a small python script that do that and output the two info you are looking at _or_ report a failure.
That would be more secured & offer options to add more detection & reporting if required later without impact the performance.

Reply:
This could be merged as-is if the performance impact isn't too high and update it later so it can be in the release.